doc/man/man3/XDS.3.man
'\" rtp
.\" vim: ft=nroff sw=4 noet nocin nosi com=b\:.\\\" fo+=tcqlorn tw=77
.\" =========================================================================
.\"
.\" @(#) doc/man/man3/XDS.3.man
.\"
.\" =========================================================================
.\"
.\" Copyright (c) 2008-2011  Monavacon Limited <http://www.monavacon.com/>
.\" Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
.\" Copyright (c) 1997-2001  Brian F. G. Bidulock <bidulock@openss7.org>
.\"
.\" All Rights Reserved.
.\"
.\" Permission is granted to copy, distribute and/or modify this manual under
.\" the terms of the GNU Free Documentation License, Version 1.3 or any later
.\" version published by the Free Software Foundation; with no Invariant
.\" Sections, no Front-Cover Texts, and no Back-Cover Texts.  A copy of the
.\" license is included in the section entitled "GNU Free Documentation
.\" License".
.\"
.\" Permission to use, copy and distribute this manual without modification,
.\" for any purpose and without fee or royalty is hereby granted, provided
.\" that both the above copyright notice and this permission notice appears
.\" in all copies and that the name of OpenSS7 Corporation not be used in
.\" advertising or publicity pertaining to distribution of this documentation
.\" or its contents without specific, written prior permission.  OpenSS7
.\" Corporation makes no representation about the suitability of this manual
.\" for any purpose.  It is provided "as is" without express or implied
.\" warranty.
.\"
.\" Permission is granted to process this file through groff and print the
.\" results, provided the printed document carries a copying permission
.\" notice identical to this one except for the removal of this paragraph
.\" (this paragraph not being relevant to the printed manual).
.\"
.\" OPENSS7 CORPORATION DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS MANUAL
.\" INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
.\" PARTICULAR PURPOSE, NON-INFRINGEMENT, OR TITLE; THAT THE CONTENTS OF THE
.\" DOCUMENT ARE SUITABLE FOR ANY PURPOSE, OR THAT THE IMPLEMENTATION OF SUCH
.\" CONTENTS WILL NOT INFRINGE ON ANY THIRD PARTY PATENTS, COPYRIGHTS,
.\" TRADEMARKS OR OTHER RIGHTS.  IN NO EVENT SHALL OPENSS7 CORPORATION BE
.\" LIABLE FOR ANY DIRECT, INDIRECT, SPECIAL OR CONSEQUENTIAL DAMAGES OR ANY
.\" DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
.\" IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING
.\" OUT OF OR IN CONNECTION WITH ANY USE OF THIS DOCUMENT OR THE PERFORMANCE
.\" OR IMPLEMENTATION OF THE CONTENTS THEREOF.
.\" 
.\" Since the Linux kernel and libraries are constantly changing, this manual
.\" page may be incorrect or out-of-date.  The author(s) assume no
.\" responsibility for errors or omissions, or for damages resulting from the
.\" use of the information contained herein.  The author(s) may not have
.\" taken the same level of care in the production of this manual, which is
.\" licensed free of charge, as they might when working professionally.  The
.\" author(s) will take no responsibility in it.
.\" 
.\" Formatted or processed versions of this manual, if unaccompanied by the
.\" source, must acknowledge the copyright and authors of this work.
.\"
.\" -------------------------------------------------------------------------
.\"
.\" U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software on
.\" behalf of the U.S. Government ("Government"), the following provisions
.\" apply to you.  If the Software is supplied by the Department of Defense
.\" ("DoD"), it is classified as "Commercial Computer Software" under
.\" paragraph 252.227-7014 of the DoD Supplement to the Federal Acquisition
.\" Regulations ("DFARS") (or any successor regulations) and the Government
.\" is acquiring only the license rights granted herein (the license rights
.\" customarily provided to non-Government users).  If the Software is
.\" supplied to any unit or agency of the Government other than DoD, it is
.\" classified as "Restricted Computer Software" and the Government's rights
.\" in the Software are defined in paragraph 52.227-19 of the Federal
.\" Acquisition Regulations ("FAR") (or any successor regulations) or, in the
.\" cases of NASA, in paragraph 18.52.227-86 of the NASA Supplement to the
.\" FAR (or any successor regulations).
.\"
.\" =========================================================================
.\" 
.\" Commercial licensing and support of this software is available from
.\" OpenSS7 Corporation at a fee.  See http://www.openss7.com/
.\" 
.\" =========================================================================
.\"
.R1
bracket-label "\fR[\fB" "\fR]" "\fR, \fB"
no-default-database
database openss7.refs
accumulate
move-punctuation
abbreviate A
join-authors ", " ", " " and "
et-al " et al" 2 3
abbreviate-label-ranges ".."
sort-adjacent-labels
search-truncate 40
search-ignore CGIQOSTU
.R2
.so openss7.macros
.\"
.\"
.TH XDS 3 "@PACKAGE_DATE@" "@PACKAGE@-@VERSION@" "@PACKAGE_TITLE@ Library Calls"
.\"
.\"
.SH NAME
.B XDS
\- API to Directory Services (XDS)
.\"
.\"
.SH SYNOPSIS
.PP
.ad l
.nh
.HP
.BI "DS_status ds_abandon(OM_private_object " session ", OM_sint " invoke_id );
.PD 0
.HP
.BI "DS_status ds_add_entry(OM_private_object " session ", OM_private_object " context ", OM_object " name ", OM_object " entry ", OM_sint *" invoke_id_return ); 
.HP
.BI "DS_status ds_bind(OM_object " session ", OM_workspace " workspace ", OM_private_object *" bound_session_return );
.HP
.BI "DS_status ds_compare(OM_private_object " session ", OM_private_object " context ", OM_object " name ", OM_object " ava ", OM_private_object " result_return ", OM_sint *" invoke_id_return );
.HP
.BI "OM_workspace ds_initialize(void);
.HP
.BI "DS_status ds_list(OM_private_object " session ", OM_private_object " context ", OM_object " name ", OM_private_object *" result_return ", OM_sint *" invoke_id_return );
.HP
.BI "DS_status ds_modify_entry(OM_private_object " session ", OM_private_object " context ", OM_object " name ", OM_object " changes ", OM_sint *" invoke_id_return );
.HP
.BI "DS_status ds_modify_rdn(OM_private_object " session ", OM_private_object " context ", OM_object " name ", OM_object " new_RDN ", OM_boolean " delete_old_RDN ", OM_sint *" invoke_id_return );
.HP
.BI "DS_status ds_read(OM_private_object " session ", OM_private_object " context ", OM_object " name ", OM_object " selection ", OM_private_object *" result_return ", OM_sint *" invoke_id_return );
.HP
.BI "DS_status ds_receive_result(OM_private_object " session ", OM_sint " invoke_id ", OM_uint *" completion_flag_return ", DS_status *" operation_status_return ", OM_private_object *" result_return ", OM_sint *" invoke_id_return );
.HP
.BI "DS_status ds_remove_entry(OM_private_object " session ", OM_private_object " context ", OM_object " name ", OM_sint *" invoke_id_return );
.HP
.BI "DS_status ds_search(OM_private_object " session ", OM_private_object " context ", OM_object " name ", OM_sint " subset ", OM_object " filter ", OM_boolean " search_aliases ", OM_object " selection ", OM_private_object *" result_return ", OM_sint *" invoke_id_return );
.HP
.BI "DS_status ds_shutdown(OM_workspace " workspace );
.HP
.BI "DS_status ds_unbind(OM_private_object " session );
.HP
.BI "DS_status ds_version(DS_feature " feature_list[] ", OM_workspace " workspace );
.PD
.ad b
.hy 1
.\"
.\"
.SH OVERVIEW
.PP
This manual page describes the OpenSS7 Standard XDS Library.
.\"
.\"
.SH DESCRIPTION
.PP
The XDS Library contains the following functions:
.\"
.\"
.SH FUNCTIONS
.PP
.TP
.BR ds_abandon (3)
.PD 0
.TP
.BR ds_add_entry (3)
.TP
.BR ds_bind (3)
.TP
.BR ds_compare (3)
.TP
.BR ds_initialize (3)
.TP
.BR ds_list (3)
.TP
.BR ds_modify_entry (3)
.TP
.BR ds_modify_rdn (3)
.TP
.BR ds_read (3)
.TP
.BR ds_receive_result (3)
.TP
.BR ds_remove_entry (3)
.TP
.BR ds_search (3)
.TP
.BR ds_shutdown (3)
.TP
.BR ds_unbind (3)
.TP
.BR ds_version (3)
.PD
.\"
.\"
.SH COMPATIBILITY
.PP
.B libxds
is compatible with XDS,
.[
xds c608
.]
and implementations based on XDS with the following portability
considerations:
.IP \(em \w'\(em\(em'u
.PP
See
.BR OSI (3)
for additional compatibility information.
.\"
.\"
.SH CONFORMANCE
.PP
XDS.
.[
xds c608
.]
.\"
.\"
.SH HISTORY
.PP
.B libxds
first appeared as
XDS.
.[
xds c608
.]
.\"
.\"
.SH VERSIONS
.PP
This manual page was written for the OpenSS7 standard version of the XDS
Library.
.\"
.\"
.XX
.[
xds c608
.]
.[
XAP c303
.]
.[
XX25
.]
.[
XOM c607
.]
.\"
.\"
.[
$LIST$
.]
.TI


NEWS
OpenSS7 -- history of user visible changes.  2015-01-15
$Id: NEWS,v 1.1.2.25 2011-05-10 13:45:30 brian Exp $
Copyright (c) 2008-2015  Monavacon Limited. <http://www.monavacon.com/>
Copyright (c) 2001-2008  OpenSS7 Corporation. <http://www.openss7.com/>
Copyright (c) 1997-2001  Brian Bidulock <bidulock@openss7.org>
See the end for copying conditions (for this file).

The sections that follow provide information on OpenSS7 releases of the
'OpenSS7' package.

Major changes for release openss7-1.1.7.20150115
------------------------------------------------

This is the seventh release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Source code release on GitHub.

Major changes for release openss7-1.1.7.20131209
------------------------------------------------

This is the seventh release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Build corrections for RHEL 6.4.

   - Scripts for country boundary, shoreline and rivers database
     preparation.

   - Database for ANSI T1.101 point code assignments.

   - Suppress extransous /proc/1/comm error messages when testing for
     systemd.

   - Handle PUIAS to Springdale rename for PUIAS distribution.

Major changes for release openss7-1.1.7.20131123
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Correction to ca-cert handling in RHEL 6 install scriptlet.

   - RHEL 6 build changes.

   - Set permissions correctly when applying kernel module patch.

   - Do not duplicate certs when including ca-cert.

   - Avoid conflicting shelll variable in RPM install scriplet.

   - Remove /proc/1/comm checks for systemd in module loader.

Major changes for release openss7-1.1.7.20130209
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Corrections to RPM spec file.

Major changes for release openss7-1.1.7.20130129
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Compensate for RHEL6 lack of c_rehash.

   - Do not create sysfs nodes: they do not work properly.

   - Use no foreground flag in strace.service file.

   - More error traces for strerr logger.

   - Remove static device names (the do not work either).

   - Changes to support weak kernel modules.

   - Clean up weak module support.

Major changes for release openss7-1.1.7.20130125
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Better Archlinux install script support.

Major changes for release openss7-1.1.6.20130125
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Remove use of kill_litter_super().

Major changes for release openss7-1.1.3.20130125
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Updates and corrections to weak kernel module builds.

   - Repo support and corrections for Archlinux packages.

Major changes for release openss7-1.1.3.20130123
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Additional support for Archlinux.

   - Add udev rules for strace and strerr.

   - Add system unit files for strace and strerr.

   - Add repo support for Archlinux packages.

   - Build Archlinux packages better.

Major changes for release openss7-1.1.3.20130121
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Build support for SLES 11.

   - Handle certificates better.

   - Get configuration files in the right places for SLES 11.

   - Support in RPM spec file for certificates on SLES 11.

   - Create a new OpenSS7 certificate.

   - Repository support for SLES 11.

   - RPM spec file reworked for new RPM release.

   - Better certificate handling for RHEL.

   - Additional build support for RHEL 6 with certificates.


src/lib/xds.c
/*****************************************************************************

 @(#) File: src/lib/xds.c

 -----------------------------------------------------------------------------

 Copyright (c) 2008-2015  Monavacon Limited <http://www.monavacon.com/>
 Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
 Copyright (c) 1997-2001  Brian F. G. Bidulock <bidulock@openss7.org>

 All Rights Reserved.

 This program is free software: you can redistribute it and/or modify it under
 the terms of the GNU Affero General Public License as published by the Free
 Software Foundation, version 3 of the license.

 This program is distributed in the hope that it will be useful, but WITHOUT
 ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
 details.

 You should have received a copy of the GNU Affero General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>, or
 write to the Free Software Foundation, Inc., 675 Mass Ave, Cambridge, MA
 02139, USA.

 -----------------------------------------------------------------------------

 U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software on
 behalf of the U.S. Government ("Government"), the following provisions apply
 to you.  If the Software is supplied by the Department of Defense ("DoD"), it
 is classified as "Commercial Computer Software" under paragraph 252.227-7014
 of the DoD Supplement to the Federal Acquisition Regulations ("DFARS") (or any
 successor regulations) and the Government is acquiring only the license rights
 granted herein (the license rights customarily provided to non-Government
 users).  If the Software is supplied to any unit or agency of the Government
 other than DoD, it is classified as "Restricted Computer Software" and the
 Government's rights in the Software are defined in paragraph 52.227-19 of the
 Federal Acquisition Regulations ("FAR") (or any successor regulations) or, in
 the cases of NASA, in paragraph 18.52.227-86 of the NASA Supplement to the FAR
 (or any successor regulations).

 -----------------------------------------------------------------------------

 Commercial licensing and support of this software is available from OpenSS7
 Corporation at a fee.  See http://www.openss7.com/

 *****************************************************************************/

static char const ident[] = "src/lib/xds.c (" PACKAGE_ENVR ") " PACKAGE_DATE;

/* This file can be processed with doxygen(1). */

/** @weakgroup xds OpenSS7 XDS Library
  * @{ */

/** @file
  * OpenSS7 X/Open Data Services (XDS) implementation file. */

/**
  * This manual contains documentation of the OpenSS7 XDS Library functions that
  * are generated automatically from the source code with doxygen.  This
  * documentation is intended to be used for maintainers of the OpenSS7 XDS
  * Library and is not intended for users of the OpenSS7 XDS Library.  Users
  * should consult the documentation found in the user manual pages beginning
  * with xds(3).
  */

#include <xds.h>

/** @brief
  * @param session
  * @param invoke_id
  * @version XDS_1.0
  * @par Alias:
  * This symbol is an implementation of __xds_ds_abandon().
  */
DS_status
__xds_ds_abandon(OM_private_object session, OM_sint invoke_id)
{
}

/** @fn DS_status ds_abandon(OM_private_object session, OM_sint invoke_id)
  * @param session
  * @param invoke_id
  * @version XDS_1.0
  * @par Alias:
  * This symbol is a strong alias of __xds_ds_abandon().
  */
__asm__(".symver __xds_ds_abandon,ds_abandon@@XDS_1.0");

/** @brief
  * @param session
  * @param context
  * @param name
  * @param entry
  * @param invoke_id_return
  * @version XDS_1.0
  * @par Alias:
  * This symbol is an implementation of __xds_ds_add_entry().
  */
DS_status
__xds_ds_add_entry(OM_private_object session, OM_private_object context, OM_object name,
		   OM_object entry, OM_sint * invoke_id_return)
{
}

/** @fn DS_status ds_add_entry(OM_private_object session, OM_private_object context, OM_object name, OM_object entry, OM_sint * invoke_id_return)
  * @param session
  * @param context
  * @param name
  * @param entry
  * @param invoke_id_return
  * @version XDS_1.0
  * @par Alias:
  * This symbol is a strong alias of __xds_ds_add_entry().
  */
__asm__(".symver __xds_ds_add_entry,ds_add_entry@@XDS_1.0");

/** @brief
  * @param session
  * @param workspace
  * @param bound_session_return
  * @version XDS_1.0
  * @par Alias:
  * This symbol is an implementation of __xds_ds_bind().
  */
DS_status
__xds_ds_bind(OM_object session, OM_workspace workspace, OM_private_object * bound_session_return)
{
}

/** @fn DS_status ds_bind(OM_object session, OM_workspace workspace, OM_private_object * bound_session_return)
  * @param session
  * @param workspace
  * @param bound_session_return
  * @version XDS_1.0
  * @par Alias:
  * This symbol is a strong alias of __xds_ds_bind().
  */
__asm__(".symver __xds_ds_bind,ds_bind@@XDS_1.0");

/** @brief
  * @param session
  * @param context
  * @param name
  * @param ava
  * @param result_return
  * @param invoke_id_return
  * @version XDS_1.0
  * @par Alias:
  * This symbol is an implementation of __xds_ds_compare().
  */
DS_status
__xds_ds_compare(OM_private_object session, OM_private_object context, OM_object name,
		 OM_object ava, OM_private_object result_return, OM_sint * invoke_id_return)
{
}

/** @fn DS_status ds_compare(OM_private_object session, OM_private_object context, OM_object name, OM_object ava, OM_private_object result_return, OM_sint * invoke_id_return)
  * @param session
  * @param context
  * @param name
  * @param ava
  * @param result_return
  * @param invoke_id_return
  * @version XDS_1.0
  * @par Alias:
  * This symbol is a strong alias of __xds_ds_compare().
  */
__asm__(".symver __xds_ds_compare,ds_compare@@XDS_1.0");

/** @brief
  * @version XDS_1.0
  * @par Alias:
  * This symbol is an implementation of _xds_ds_initialize().
  */
OM_workspace
__xds_ds_initialize(void)
{
}

/** @fn OM_workspace ds_initialize(void)
  * @version XDS_1.0
  * @par Alias:
  * This symbol is a strong alias of _xds_ds_initialize().
  */
__asm__(".symver __xds_ds_initialize,ds_initialize@@XDS_1.0");

/** @brief
  * @param session
  * @param context
  * @param name
  * @param result_return
  * @param invoke_id_return
  * @version XDS_1.0
  * @par Alias:
  * This symbol is an implementation of __xds_ds_list().
  */
DS_status
__xds_ds_list(OM_private_object session, OM_private_object context, OM_object name,
	      OM_private_object * result_return, OM_sint * invoke_id_return)
{
}

/** @fn DS_status ds_list(OM_private_object session, OM_private_object context, OM_object name, OM_private_object * result_return, OM_sint * invoke_id_return)
  * @param session
  * @param context
  * @param name
  * @param result_return
  * @param invoke_id_return
  * @version XDS_1.0
  * @par Alias:
  * This symbol is a strong alias of __xds_ds_list().
  */
__asm__(".symver __xds_ds_list,ds_list@@XDS_1.0");

/** @brief
  * @param session
  * @param context
  * @param name
  * @param changes
  * @param invoke_id_return
  * @version XDS_1.0
  * @par Alias:
  * This symbol is an implementation of __xds_ds_modify_entry().
  */
DS_status
__xds_ds_modify_entry(OM_private_object session, OM_private_object context, OM_object name,
		      OM_object changes, OM_sint * invoke_id_return)
{
}

/** @fn DS_status ds_modify_entry(OM_private_object session, OM_private_object context, OM_object name, OM_object changes, OM_sint * invoke_id_return)
  * @param session
  * @param context
  * @param name
  * @param changes
  * @param invoke_id_return
  * @version XDS_1.0
  * @par Alias:
  * This symbol is a strong alias of __xds_ds_modify_entry().
  */
__asm__(".symver __xds_ds_modify_entry,ds_modify_entry@@XDS_1.0");

/** @brief
  * @param session
  * @param context
  * @param name
  * @param new_RDN
  * @param delete_old_RDN
  * @param invoke_id_return
  * @version XDS_1.0
  * @par Alias:
  * This symbol is an implementation of __xds_ds_modify_rdn().
  */
DS_status
__xds_ds_modify_rdn(OM_private_object session, OM_private_object context, OM_object name,
		    OM_object new_RDN, OM_boolean delete_old_RDN, OM_sint * invoke_id_return)
{
}

/** @fn DS_status ds_modify_rdn(OM_private_object session, OM_private_object context, OM_object name, OM_object new_RDN, OM_boolean delete_old_RDN, OM_sint * invoke_id_return)
  * @param session
  * @param context
  * @param name
  * @param new_RDN
  * @param delete_old_RDN
  * @param invoke_id_return
  * @version XDS_1.0
  * @par Alias:
  * This symbol is a strong alias of __xds_ds_modify_rdn().
  */
__asm__(".symver __xds_ds_modify_rdn,ds_modify_rdn@@XDS_1.0");

/** @brief
  * @param session
  * @param context
  * @param name
  * @param selection
  * @param result_return
  * @param invoke_id_return
  * @version XDS_1.0
  * @par Alias:
  * This symbol is an implementation of __xds_ds_read().
  */
DS_status
__xds_ds_read(OM_private_object session, OM_private_object context, OM_object name,
	      OM_object selection, OM_private_object * result_return, OM_sint * invoke_id_return)
{
}

/** @fn DS_status ds_read(OM_private_object session, OM_private_object context, OM_object name, OM_object selection, OM_private_object * result_return, OM_sint * invoke_id_return)
  * @param session
  * @param context
  * @param name
  * @param selection
  * @param result_return
  * @param invoke_id_return
  * @version XDS_1.0
  * @par Alias:
  * This symbol is a strong alias of __xds_ds_read().
  */
__asm__(".symver __xds_ds_read,ds_read@@XDS_1.0");

/** @brief
  * @param session
  * @param invoke_id
  * @param completion_flag_return
  * @param operation_status_return
  * @param result_return
  * @param invoke_id_return
  * @version XDS_1.0
  * @par Alias:
  * This symbol is an implementation of __xds_ds_receive_result().
  */
DS_status
__xds_ds_receive_result(OM_private_object session, OM_sint invoke_id,
			OM_uint * completion_flag_return, DS_status * operation_status_return,
			OM_private_object * result_return, OM_sint * invoke_id_return)
{
}

/** @fn DS_status ds_receive_result(OM_private_object session, OM_sint invoke_id, OM_uint * completion_flag_return, DS_status * operation_status_return, OM_private_object * result_return, OM_sint * invoke_id_return)
  * @param session
  * @param invoke_id
  * @param completion_flag_return
  * @param operation_status_return
  * @param result_return
  * @param invoke_id_return
  * @version XDS_1.0
  * @par Alias:
  * This symbol is a strong alias of __xds_ds_receive_result().
  */
__asm__(".symver __xds_ds_receive_result,ds_receive_result@@XDS_1.0");

/** @brief
  * @param session
  * @param context
  * @param name
  * @param invoke_id_return
  * @version XDS_1.0
  * @par Alias:
  * This symbol is an implementation of __xds_ds_remove_entry().
  */
DS_status
__xds_ds_remove_entry(OM_private_object session, OM_private_object context, OM_object name,
		      OM_sint * invoke_id_return)
{
}

/** @fn DS_status ds_remove_entry(OM_private_object session, OM_private_object context, OM_object name, OM_sint * invoke_id_return)
  * @param session
  * @param context
  * @param name
  * @param invoke_id_return
  * @version XDS_1.0
  * @par Alias:
  * This symbol is a strong alias of __xds_ds_remove_entry().
  */
__asm__(".symver __xds_ds_remove_entry,ds_remove_entry@@XDS_1.0");

/** @brief
  * @param session
  * @param context
  * @param name
  * @param subset
  * @param filter
  * @param search_aliases
  * @param selection
  * @param result_return
  * @param invoke_id_return
  * @version XDS_1.0
  * @par Alias:
  * This symbol is an implementation of __xds_ds_search().
  */
DS_status
__xds_ds_search(OM_private_object session, OM_private_object context, OM_object name,
		OM_sint subset, OM_object filter, OM_boolean search_aliases,
		OM_object selection, OM_private_object * result_return, OM_sint * invoke_id_return)
{
}

/** @fn DS_status ds_search(OM_private_object session, OM_private_object context, OM_object name, OM_sint subset, OM_object filter, OM_boolean search_aliases, OM_object selection, OM_private_object * result_return, OM_sint * invoke_id_return)
  * @param session
  * @param context
  * @param name
  * @param subset
  * @param filter
  * @param search_aliases
  * @param selection
  * @param result_return
  * @param invoke_id_return
  * @version XDS_1.0
  * @par Alias:
  * This symbol is a strong alias of __xds_ds_search().
  */
__asm__(".symver __xds_ds_search,ds_search@@XDS_1.0");

/** @brief
  * @param workspace
  * @version XDS_1.0
  * @par Alias:
  * This symbol is an implementation of __xds_ds_shutdown().
  */
DS_status
__xds_ds_shutdown(OM_workspace workspace)
{
}

/** @fn DS_status ds_shutdown(OM_workspace workspace)
  * @param workspace
  * @version XDS_1.0
  * @par Alias:
  * This symbol is strong alias of __xds_ds_shutdown().
  */
__asm__(".symver __xds_ds_shutdown,ds_shutdown@@XDS_1.0");

/** @brief
  * @param session
  * @version XDS_1.0
  * @par Alias:
  * This symbol is an implementation of __xds_ds_unbind().
  */
DS_status
__xds_ds_unbind(OM_private_object session)
{
}

/** @fn DS_status ds_unbind(OM_private_object session)
  * @param session
  * @version XDS_1.0
  * @par Alias:
  * This symbol is a strong alias of __xds_ds_unbind().
  */
__asm__(".symver __xds_ds_unbind,ds_unbind@@XDS_1.0");

/** @brief
  * @param feature_list
  * @param workspace
  * @version XDS_1.0
  * @par Alias:
  * This symbol is an implementation of __xds_ds_version().
  */
DS_status
__xds_ds_version(DS_feature feature_list[], OM_workspace workspace)
{
}

/** @fn DS_status ds_version(DS_feature feature_list[], OM_workspace workspace)
  * @param feature_list
  * @param workspace
  * @version XDS_1.0
  * @par Alias:
  * This symbol is a strong alias of __xds_ds_version().
  */
__asm__(".symver __xds_ds_version,ds_version@@XDS_1.0");

/** @} */

// vim: com=srO\:/**,mb\:*,ex\:*/,srO\:/*,mb\:*,ex\:*/,b\:TRANS
